### PR TITLE
Adding the doc for datagen source connector

### LINKED ISSUE #268: Document datagen source connector
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

It is missing from the page https://www.risingwave.dev/docs/latest/sql-create-source/

### Reference

Checkout the example here:https://github.com/risingwavelabs/risingwave/blob/main/e2e_test/streaming/datagen.slt

### LINKED ISSUE #371: Website: adjust table headings in multi-column tables
### Where's the bug?

https://github.com/risingwavelabs/risingwave-docs/pull/370

### Describe the bug

The table needs a tiny adjustment.
The heading of the second column now exceeds its content when the table has multiple columns.
Please fix it thanks.
image
image
image

### PR DESCRIPTION
**Pull request by CharlieSYH**
## Info

Preview: https://pr-370.d2fbku9n2b6wde.amplifyapp.com/docs/latest/create-source-datagen/

@lmatz @neverchanje @tabVersion Some information is still missing. Please help provide the following:
- The default value for `fields.column_name.seed`;
- The range for each parameter if necessary.

@Ad-Bean The table needs a tiny adjustment. The heading of the second column now exceeds the content when the table has multiple columns. Please fix it thanks.
image
image
image

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/268

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

**Comment by CharlieSYH** [comment]
Thank you for making the table better, @Ad-Bean !

**Comment by tabVersion** [comment]
> The default value for fields.column_name.seed

To avoid all generators producing the same sequence and to ensure that this sequence is reproducible, we use the serial number of each slice as the seed.

For example, there are three slices in this source, then they will use 0, 1, and 2 as seeds respectively.

File: docs/sql/create-source/create-source-datagen.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+<Tabs>
+<TabItem value="sequence" label="Sequence">
+
+The sequence load generator can generate numbers, incremented by 1, from the starting number to the ending number. For example, `1`, `2`, `3`, ... and `1.56`, `2.56`, `3.56`, ...
+
+Specify the following fields for every column.
+
+|Field|Description|Value|Required?|
+|---|---|---|---|
+|fields.*column_name*.kind|Generator type|Set to `sequence`.|True|
```
Comment (tabVersion) [inline docs/sql/create-source/create-source-datagen.md]: if `fields.*column_name*.kind` is not specified to `sequence`, it will generate random value

**Comment by tabVersion** [comment]
For datagen, every data type has its own options

* `DataType::Timestamp`: randomly generate a timestamp from `fields.*column_name*.max_past` (default 1 day) to now. Note that this data type does not support sequence generating. 
* `DataType::Varchar`: randomly generate a string, whose length is specified by `fields.*column_name*.length` (default 10). Note that this data type does not support sequence generating. 
* `DataType::Struct`: a combiner of other fields
* other types
  * if specified to `sequence`: will generate a sequence from `fields.*column_name*.start` (default 0) to `fields.*column_name*.end` (default i16::MAX)
  * if not, will generate random values from `fields.*column_name*.min` (default 0) to `fields.*column_name*.max` (default i16::MAX), including both bounds of the range, eg. `min <= value <= max`.

File: docs/sql/create-source/create-source-datagen.md (hunk 1 above)
Comment (CharlieSYH) [reply docs/sql/create-source/create-source-datagen.md]: image

**Comment by CharlieSYH** [comment]
@tabVersion Please check out the latest draft: https://pr-370.d2fbku9n2b6wde.amplifyapp.com/docs/latest/create-source-datagen/

File: docs/sql/create-source/create-source-datagen.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+   connector = ' datagen ',
+   column_parameter = ' value ', ...  -- Configure the generator for each column. See detailed information below.
+   datagen.rows.per.second = ' rows_integer '  -- Specify how many rows of records to generate every second. For example, '20'.
+) 
+ROW FORMAT JSON;
+```
+
+### `WITH` options - *`column_parameter`*
+
+The following table shows the available load generator types for each data type.
```
Comment (hengm3467) [inline docs/sql/create-source/create-source-datagen.md]: Not sure if this makes more sense: "shows the data types can be generated for each load generator type."

File: docs/sql/create-source/create-source-datagen.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+   The random varchar generator produces random combination of uppercase and lowercase letters and numbers.
+   
+   Specify the following fields for every column in the source you are creating.
+
+   |Field|Description|Value|Required?|
+   |---|---|---|---|
+   |fields.*column_name*.kind|Generator type|Set to `random`.|False<br/>Default: `random`|
+   |fields.*column_name*.length|The length of the varchar to be generated.|A positive integer<br/>Example: `100`|False<br/>Default: `10`|
+   |fields.*column_name*.seed|Seed number initializes the random load generator. The sequence of the generated numbers is determined by the seed value. If given the same seed number, the genrator will produce the same sequence of numbers.|A positive integer<br/>Example: `3`|False<br/>If not specified, the generator will produce the same sequence of characters if the length is unchanged.|
```
Comment (tabVersion) [inline docs/sql/create-source/create-source-datagen.md]: > If not specified

It looks weird to me. What we want to express is that the sequence generated with constant seed and sequence length is fixed.

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: docs/sql/create-source/create-source-datagen.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+|Sequence|✓|✕|✕|
+|Random|✓|✓|✓|
+
+Select a generator type:
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+<TabItem value="sequence" label="Sequence">
```
Comment (hengm3467) [inline docs/sql/create-source/create-source-datagen.md:39]: Do we want to capitalize Sequence in the label?

File: docs/sql/create-source/create-source-datagen.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+Specify the following fields for every column.
+
+|Field|Description|Value|Required?|
+|---|---|---|---|
+|fields.*column_name*.kind|Generator type|Set to `sequence`.|False<br/>Default: `random`|
+|fields.*column_name*.start|Starting number<br/>Must be smaller than the ending number.|Any number of the column data type<br/>Example: `50`|False<br/>Default: `0`|
+|fields.*column_name*.end|Ending number<br/>Must be larger than the starting number.|Any number of the column data type<br/>Example: `100`|False<br/>Default: `32767`|
+
+</TabItem>
+<TabItem value="random" label="Random">
```
Comment (hengm3467) [inline docs/sql/create-source/create-source-datagen.md:52]: Same. Do we want to capitalize "Random" here?

File: docs/sql/create-source/create-source-datagen.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+   The random timestamp generator produces random timestamp earlier than the current date and time.
+
+   Specify the following fields for every column in the source you are creating.
+   
+   |Field|Description|Value|Required?|
+   |---|---|---|---|
+   |fields.*column_name*.kind|Generator type|Set to `random`.|False<br/>Default: `random`|
+   |fields.*column_name*.max_past|Specify the maximum time interval to determine the earliest timestamp can be generated.|An [interval](../sql-data-types.md)<br/>Example: `2h 37min`|False<br/>Default: `1 day`|
+   |fields.*column_name*.seed|Seed number initializes the random load generator. The sequence of the generated numbers is determined by the seed value. If given the same seed number, the generator will produce the same sequence of numbers.|A positive integer<br/>Example: `3`|False<br/>If not specified, the genrator will produce the same sequence of timestamps if the system time and interval is unchanged.|
```
Comment (hengm3467) [inline docs/sql/create-source/create-source-datagen.md]: -> A seed number that initializes...

File: docs/sql/create-source/create-source-datagen.md (hunk 3 above)
Comment (hengm3467) [inline docs/sql/create-source/create-source-datagen.md]: -> A seed number that initializes...

**Pull request merged**